src/resources/views/crud/filters/date_range.blade.php

```diff
@@ -90,31 +142,65 @@ function applyDateRangeFilter{{camel_case($filter->name)}}(start, end) {
 			// mark this filter as active in the navbar-filters
 			if (URI(new_url).hasQuery('{{ $filter->name }}', true)) {
 				$('li[filter-name={{ $filter->name }}]').removeClass('active').addClass('active');
-			}
-			else
-			{
+			} else {
 				$('li[filter-name={{ $filter->name }}]').trigger('filter:clear');
 			}
   		}
 
 		jQuery(document).ready(function($) {
-			var dateRangeInput = $('#daterangepicker-{{ str_slug($filter->name) }}').daterangepicker({
-				timePicker: false,
-		        ranges: {
-		            'Today': [moment().startOf('day'), moment().endOf('day')],
-		            'Yesterday': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
-		            'Last 7 Days': [moment().subtract(6, 'days'), moment()],
-		            'Last 30 Days': [moment().subtract(29, 'days'), moment()],
-		            'This Month': [moment().startOf('month'), moment().endOf('month')],
-		            'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
-		        },
-				@if ($filter->currentValue)
-		        startDate: moment("{{ $start_date }}"),
-		        endDate: moment("{{ $end_date }}"),
-				@endif
-				alwaysShowCalendars: true,
-				autoUpdateInput: true
-			});
+			var dateRangeInput = $('#daterangepicker-{{ str_slug($filter->name) }}');
+
+            $config = dateRangeInput.data('bs-datepicker');
+
+            $ranges = $config.ranges;
+            $config.ranges = {};
+
+            //the use of eval() is not always recommended, but this use case fits
+            //the string beeing eval'd comes from a trusty source because is always hardcoded (dev or us).
+            for (var key in $ranges) {
+                if ($ranges.hasOwnProperty(key)) {
+                    $config.ranges[key] = eval($ranges[key]);
```

Comment:
Hmm... I'm a bit uncomfortable in using ```eval()``` - as you must have guessed from the comment 😆 Let's see if we find a way around this. 

We only _need_ to use ```eval()``` because we're defining JS inside the PHP array, right? Because inside the PHP array we have values like ```moment().startOf('day'), moment().endOf('day')```. But. Why would the developer prefer to define dates using moment, instead of Carbon? After all, Backpack was built for developers who enjoy writing PHP more than they enjoy writing JS.

So what if, above where we define the PHP defaults, instead of passing:
```php
trans('backpack::crud.today') =>  "[moment().startOf('day'), moment().endOf('day')]",
```
we do this instead:
```php
trans('backpack::crud.today') =>  [Carbon::now()->startOfDay(), Carbon::now()->endOfDay()],
```
and stuff like that, for all the defaults. Basically defining them in PHP with Carbon, instead of in JS, with momentJS.

Then in JS, instead of having to use ```eval()``` we could just do a ```foreach``` through the configuration array, and turn all datetime strings into moment() objects.

I think this would be:
- safer to do (from a coding perspective)
- easier to read/understand
- easier to define (I personally would prefer to define things with Carbon rather than momentJS)
- solve the problem here https://github.com/Laravel-Backpack/CRUD/pull/2500/files where the configuration is both in JS and PHP (that's _no bueno_); again, that PR is just an example for the PHP array;

What do you think @pxpm ? Wouldn't that be better?